hangman.py

```diff
@@ -83,63 +77,170 @@ def get_available_letters(letters_guessed):
+            if user_guess in letters_guessed:
+                warnings += 1
+                if warnings < 4:
+                    print('Oops! You\'ve already guessed that letter.You now have ' + str(3 - warnings) +
+                          ' warnings left:' + get_guessed_word(secret_word, letters_guessed))
+                elif warnings == 4:
+                    print(
+                        'Oops! You\'ve already guessed that letter. You now have no warnings left so you lose one guess:'
+                        + get_guessed_word(secret_word, letters_guessed))
+                    guesses += 1
+                else:
+                    break
+
+            else:
+                letters_guessed.append(user_guess)
+                if is_word_guessed(user_guess, secret_word):
+                    print('Good guess:', get_guessed_word(secret_word, letters_guessed))
+                    if get_guessed_word(secret_word, letters_guessed) == secret_word:
+                        print('------------\nCongratulations, you won!\n'
+                              'Your total score for this game is:' + (len(set(secret_word)) * (6 - guesses)))
+                        break
+
+                else:
+                    if user_guess in ['a', 'e', 'i', 'o', 'u']:
+                        print('Oops! That letter is not in my word:' + get_guessed_word(secret_word, letters_guessed))
+                        guesses += 2
+                    else:
+                        print('Oops! That letter is not in my word:\n'
+                              'Please guess a letter:' + get_guessed_word(secret_word, letters_guessed))
+                        guesses += 1
+
+        else:
+            warnings += 1
+            if warnings < 3:
+                print('Oops! That is not a valid letter. You have ' + str(3 - warnings) + ' warnings left:'
+                      + get_guessed_word(secret_word, letters_guessed))
+            else:
+                break
+
+        if guesses == 6:
+            print('------------\n'
+                  'Sorry, you ran out of guesses.'
+                  ' The word was' + secret_word)
 
 
 def match_with_gaps(my_word, other_word):
     '''
     my_word: string with _ characters, current guess of secret word
     other_word: string, regular English word
-    returns: boolean, True if all the actual letters of my_word match the 
+    returns: boolean, True if all the actual letters of my_word match the
         corresponding letters of other_word, or the letter is the special symbol
         _ , and my_word and other_word are of the same length;
-        False otherwise: 
+        False otherwise:
     '''
-    # FILL IN YOUR CODE HERE AND DELETE "pass"
-    pass
 
+    num = 0
+    my_word = my_word.replace(' ', '')
+    if len(my_word) == len(other_word):
+        for i in range(len(my_word)):
+            if my_word[i] == '_' or my_word[i] == other_word[i]:
+                pass
```

Comment:
Operation `pass `don't uses such as a `break` in cycles. `pass` created for mute all function.